sections/platter-product-gallery.liquid

```diff
@@ -0,0 +1,118 @@
+          </div>
+          <p class="text-gray-500">{{ block.settings.product.price | money }}</p>
+        </div>
+      </article>
+    {% endfor %}
+  </div>
+
+  <footer class="flex justify-center items-center">
+    <button id="show-more" class="w-10/12 h-16 rounded-full bg-gray-800 text-white px-6 py-2 rounded mt-4 md:hidden">Show More</button>
+    <button id="show-less" class="w-10/12 h-16 rounded-full bg-gray-800 text-white px-6 py-2 rounded mt-4 hidden md:hidden">Show Less</button>
+  </footer>
+</section>
+
+<!-- The following cloudflare CDN is being used to create a static "Stars rating" icons within the product cards. -->
+<link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0-beta3/css/all.min.css" rel="stylesheet">
+
+<script type="module">
+  document.addEventListener('DOMContentLoaded', () => {
+    const mobileWidth = 768; // Define mobile width threshold, to be in sync with TailwindCSS md: breakpoint.
+
+    const toggleCardsVisibility = (isVisible) => {
+      const cards = document.querySelectorAll('.js-mobile-hide');
+      cards.forEach((card, index) => {
+        if (window.innerWidth <= mobileWidth && index > 3 && !isVisible) {
+          card.classList.add('hidden');
+        } else {
+          card.classList.remove('hidden');
+        }
+      });
+    };
+
+    const toggleVisibilityOnResize = () => {
+      window.addEventListener('resize', () => toggleCardsVisibility(false));
+    };
+
+    const handleShowMoreOrLessButtons = () => {
+      const showMoreButton = document.getElementById('show-more');
+      const showLessButton = document.getElementById('show-less');
+
+      showMoreButton.addEventListener('click', () => {
+        toggleCardsVisibility(true);
+        showMoreButton.classList.add('hidden');
+        showLessButton.classList.remove('hidden');
+      });
+
+      showLessButton.addEventListener('click', () => {
+        toggleCardsVisibility(false);
+        showLessButton.classList.add('hidden');
+        showMoreButton.classList.remove('hidden');
+      });
+    };
+
+    toggleCardsVisibility(false);
+    toggleVisibilityOnResize();
+    handleShowMoreOrLessButtons();
+  });
+</script>
+
+
+{% schema %}
+  {
+    "name": "Featured Products",
+    "class": "flex justify-center",
+    "settings": [
+      {
+        "type": "text",
+        "id": "heading",
+        "label": "Section Heading",
+        "default": "Best Sellers"
+      }
+    ],
+    "blocks": [
+      {
+        "type": "product",
+        "name": "Product Card",
+        "settings": [
+          {
+            "type": "product",
+            "id": "product",
+            "label": "Product"
+          },
+          {
+            "type": "text",
+            "id": "button_text",
+            "label": "Button Text",
+            "default": "Buy Now"
+          }
+        ],
+        "limit": 10
+      }
+    ],
+    "presets": [
+      {
+        "name": "Featured Products"
+      }
+    ]
+  }
```

Comment:
As I said above, probably there's many more section and block settings we could be adding.
But not the purpose of this technical test.